Policy Pack for Azure CIS v2.0 Enforcement for Section 9 - AppService. (#747)
turbot/guardrails-samples: cis_enforcements/azure/cis_v200/section9/README.md

---
categories: ["cis"]
---

> # Enforce Azure CIS v2.0.0 - Section 9 - AppService
> 
> This section covers security recommendations for Azure AppService.

Reviewer: ```suggestion
# Enforce Azure CIS v2.0.0 - Section 9 - App Service

This section covers security recommendations for Azure App Service.
```

I can see the CIS doc mentions `AppService`. But let's follow the actual Azure docs and use `App Service`.

```

Run the Terraform to create the policy pack in your workspace:

```sh
terraform init
terraform plan
```

Then apply the changes:

```sh
terraform apply
```

### Apply Policy Pack

Log into your Guardrails workspace and [attach the policy pack to a resource](https://turbot.com/guardrails/docs/guides/working-with-folders/smart#attach-a-smart-folder-to-a-resource).

### Enable Enforcement

> [!TIP]
> You can also update the policy settings in this policy pack directly in the Guardrails console.
>
> Please note your Terraform state file will then become out of sync and the policy settings should then only be managed in the console.
> By default, the policies are set to `Check` in the pack's policy settings. To enable automated enforcements, you can switch these policies settings by adding a comment to the `Check` setting and removing the comment from one of the listed enforcement options:

```hcl
resource "turbot_policy_setting" "azure_appservice_webapp_approved" {
  resource = turbot_smart_folder.main.id
  type     = "tmod:@turbot/azure-appservice#/policy/types/webAppApproved"
  note     = "Azure CIS v2.0.0 - Control: 9.1, 9.6, 9,7 and 9.8"
  value    = "Check: Approved"

>   # value    = "Enforce: Delete unapproved"

Reviewer: ```suggestion
  # value    = "Enforce: Delete unapproved if new"
```

> }

Reviewer: ```suggestion
}

resource "turbot_policy_setting" "azure_appservice_web_app_https_only" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-appservice#/policy/types/webAppHttpsOnly"
  note         = "Azure CIS v2.0.0 - Control: 9.2"
  # value        = "Check: Enabled"
  value        = "Enforce: Enabled"
}

resource "turbot_policy_setting" "azure_appservice_web_app_minimum_tls_version" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-appservice#/policy/types/webAppMinimumTlsVersion"
  note         = "Azure CIS v2.0.0 - Control: 9.3"
  # value        = "Check: TLS 1.2"
  value        = "Enforce: TLS 1.2"
}

resource "turbot_policy_setting" "azure_appservice_web_app_http20_enabled" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-appservice#/policy/types/webAppHttp20Enabled"
  note         = "Azure CIS v2.0.0 - Control: 9.9"
  # value        = "Check: Enabled"
  value        = "Enforce: Enabled"
}

resource "turbot_policy_setting" "azure_appservice_web_app_ftps_state" {
  resource = turbot_smart_folder.main.id
  type         = "tmod:@turbot/azure-appservice#/policy/types/webAppFtpsState"
  note         = "Azure CIS v2.0.0 - Control: 9.10"
  # value        = "Check: FTPS only"
  value        = "Enforce: FTPS only"
}
```
